webcomponents/custom.html

```diff
@@ -51,25 +51,43 @@
                 "keypress" : "handleKeyPress"
             },
             toggleLongDesc: function(e) {
+                var curName = $(this).attr("dg-name");
+                var $context;
+                var handleFocus = true;
+                if(curName === "*"){
+                    //All elements on the page.
+                    handleFocus=false;
+                    //Don't use document, because dilter doesn't actually do what you'd expect.
+                    //Instead, filter by all elements with dg-name
+                    $context = $("[dg-name]");
+                }
+                else 
+                {
+                    $context = $("[dg-name=" + curName + "]");
+                }
+                var hideLongdesc;
                 if($(this).attr("aria-pressed") == "true"){
-                    $(this).attr("aria-pressed", "false");
+                    $context.filter("dg-btn-longdesc").attr("aria-pressed", "false");
+                    hideLongdesc = true;
                 }
                 else{
-                    $(this).attr("aria-pressed", "true");
-                    $("dg-btn-simpledesc [").attr("aria-pressed", "false");
-                    $("dg-btn-summary").attr("aria-pressed", "false");
-                    $("dg-btn-tactiledesc").attr("aria-pressed", "false");
+                    $context.filter("dg-btn-longdesc").attr("aria-pressed", "true");
+                    hideLongdesc = false;
+                    $context.filter("dg-btn-simpledesc").attr("aria-pressed", "false");
+                    $context.filter("dg-btn-summary").attr("aria-pressed", "false");
+                    $context.filter("dg-btn-tactiledesc").attr("aria-pressed", "false");
                 }
-                $('dg-summary').attr('style', 'display: none;');
-                $('dg-simpledesc').attr('style', 'display: none;');
-                $('dg-tactiledesc').attr('style', 'display: none;');
-                var el = $('dg-longdesc');
-                if (el.attr('style') == undefined) {
+                $context.filter('dg-summary').attr('style', 'display: none;');
+                $context.filter('dg-simpledesc').attr('style', 'display: none;');
+                $context.filter('dg-tactiledesc').attr('style', 'display: none;');
+                var el = $context.filter('dg-longdesc');
+                if (hideLongdesc) {
                     el.attr('style', 'display: none;');
                 }
                 else {
                     el.removeAttr('style');
-                    $('#long_desc').focus();
+                    if(handleFocus)
+                        el.find('div').focus();
```

Comment:
Good point. Maybe we should set a id on the heading to focus.


On 7/6/2016 12:03 PM, Charles LaPierre wrote:
>
> In webcomponents/custom.html 
> <https://github.com/benetech/dgmr-rdm-web/pull/2#discussion_r69780654>:
>
> >                      el.attr('style', 'display: none;');
> >                  }
> >                  else {
> >                      el.removeAttr('style');
> > -                    $('#long_desc').focus();
> > +                    if(handleFocus)
> > +                        el.find('div').focus();
>
> Is this the best option here to find the 'div' to focus? what happens 
> if there are multiple divs on this page how will it know which div to 
> focus on?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/benetech/dgmr-rdm-web/pull/2/files/7c54953b7c6bb7cf3fc15ec8192d07dc908402e5..e869d1b138bcc26becf49507d25b69c56e0e8e26#r69780654>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AFGivWC36slbseN6izxGY6Uv9YxXylQYks5qS-3ogaJpZM4JGVBf>.
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194